File: mobile/src/main/java/jp/satorufujiwara/kotlin/data/repository/RepositoryModule.java
package jp.satorufujiwara.kotlin.data.repository;

import dagger.Module;
import dagger.Provides;
import jp.satorufujiwara.kotlin.AppScope;
import jp.satorufujiwara.kotlin.data.api.GitHubService;

@Module
public class RepositoryModule {

    @Provides
    @AppScope
    GitHubRepository provideGitHubRepository(GitHubService gitHubService) {
        return new GitHubRepository(gitHubService);
    }
}

File: mobile/src/main/kotlin/jp/satorufujiwara/kotlin/data/repository/RepositoryModule.kt
package jp.satorufujiwara.kotlin.data.repository

import dagger.Module
import dagger.Provides
import jp.satorufujiwara.kotlin.AppScope
import jp.satorufujiwara.kotlin.data.api.GitHubService

@Module
class RepositoryModule {

    @Provides
    @AppScope
    fun provideGitHubRepository(gitHubService: GitHubService): GitHubRepository =
            GitHubRepository(gitHubService)
}
